Frameworks/MQTTnet.AspnetCore/ApplicationBuilderExtensions.cs

```diff
@@ -17,12 +17,11 @@ public static IApplicationBuilder UseMqttEndpoint(this IApplicationBuilder app,
                 {
                     string subprotocol = null;
 
-                    if (context.Request.Headers.TryGetValue("Sec-WebSocket-Protocol", out var requestedSubProtocolValues)
-                     && requestedSubProtocolValues.Count > 0
-                     && requestedSubProtocolValues.Any(v => v.ToLower() == "mqtt")
-                     )
+                    if (context.Request.Headers.TryGetValue("Sec-WebSocket-Protocol", out var requestedSubProtocolValues))
                     {
-                        subprotocol = "mqtt";
+                        subprotocol = requestedSubProtocolValues
+                                .OrderByDescending(p => p.Length)
```

Comment:
Why ordering?

Reply:
The idea was trying to match the more specialized sub protocol eg mqttv-3.1 vs  mqttv-3.11